src/sdk/pynni/nni/compression/torch/pruning/one_shot.py

```diff
@@ -128,9 +139,42 @@ def validate_config(self, model, config_list):
         if len(config_list) > 1:
             logger.warning('Slim pruner only supports 1 configuration')
 
+
 class _StructuredFilterPruner(OneshotPruner):
-    def __init__(self, model, config_list, pruning_algorithm, optimizer=None, **algo_kwargs):
-        super().__init__(model, config_list, pruning_algorithm=pruning_algorithm, optimizer=optimizer, **algo_kwargs)
+    """
+    _StructuredFilterPruner has two ways to calculate the masks
+    for conv layers. In the normal way, the _StructuredFilterPruner
+    will calculate the mask of each layer separately. For example, each
+    conv layer determine which filters should be pruned according to its L1
+    norm. In constrast, in the dependency-aware way, the layers that in a
+    dependency group will be pruned jointly and these layers will be forced
+    to prune the same channels.
+    """
```

Comment:
suggest to put the logic of dependency awared mask generation into a new class, for example, _ConstrainedStructuredFilterPruner and overwrite `update_mask` of `Pruner`

Reply:
The old version used to have a brand new class called _ConstrainedStructuredFilterPruner, and all the constrained one-shot pruners inherit from this class. But this makes it challenging for dependency-aware pruners and original pruners to share the same set of user interfaces.(https://github.com/microsoft/nni/pull/2657#discussion_r462798140). After a discussion with @chicm-ms, we think that the interfaces of the two types of pruner should be merged upwards. We can set up a quick sync and discuss this issue~

Reply:
we can discuss today